streaming/src/main/scala/org/apache/spark/streaming/scheduler/StreamingListener.scala

```diff
@@ -66,6 +69,9 @@ case class StreamingListenerReceiverStopped(receiverInfo: ReceiverInfo)
 @DeveloperApi
 trait StreamingListener {
 
+  /** Called when the streaming starts */
```

Comment:
I don't know exactly what you're trying to capture here, but if it is the time when the `start()` method of the streaming context is called, it should be pretty trivial to get that by, huh, getting the current time in the `start()` method, where you send the event, instead of doing it in the constructor.